### PR TITLE
Using injector in callable rule

### PR DESCRIPTION
**Pull request by xepozz**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Fixed issues  | #113

### CODE REVIEW

File: src/Exception/CallbackRuleException.php [hunk 1]
```diff
@@ -10,9 +10,9 @@
 use function get_class;
 use function is_object;
 
-class CallbackRuleException extends \Exception implements FriendlyExceptionInterface
+final class CallbackRuleException extends \Exception implements FriendlyExceptionInterface
 {
-    public function __construct($result, $code = 0, Throwable $previous = null)
+    public function __construct($result, int $code = 0, Throwable $previous = null)
```
Comment (root-aza) [inline src/Exception/CallbackRuleException.php:15]: ?Throwable $previous = null

File: composer.json [hunk 2]
```diff
@@ -36,7 +36,9 @@
         "phpunit/phpunit": "^9.5",
         "roave/infection-static-analysis-plugin": "^1.7",
         "spatie/phpunit-watcher": "^1.23",
-        "vimeo/psalm": "^4.3"
+        "vimeo/psalm": "^4.3",
+        "yiisoft/injector": "^1.0",
```
Comment (samdark) [inline composer.json:40]: This one should be in `require`, not `require-dev`.

File: src/Rule/Callback.php [hunk 3]
```diff
@@ -39,4 +43,11 @@ protected function validateValue($value, ValidationContext $context = null): Res
         }
         return $result;
     }
+
+    public function withInjector(Injector $injector): self
```
Comment (samdark) [inline src/Rule/Callback.php:47]: Is there a way to avoid manually passing injector? It's not too bad. Just asking.

**Review by samdark — CHANGES_REQUESTED** [review]

File: src/Rule/Callback.php (hunk 3 above)
Comment (xepozz) [reply src/Rule/Callback.php:47]: I did some research I got that we can have one more variant when Injector will be passed to Rule through Validator. 
It looks stranger than manual passing.

File: composer.json (hunk 2 above)
Comment (xepozz) [reply composer.json:40]: If you don't need Injector you don't need to install it.

**Comment by samdark** [comment]
Discussed it. Decided not to have it because if you need to pass injector it's better to pass dependecies explicitly.

**Comment by samdark** [comment]
Applied exception improvemnts 👍

**Pull request closed without merging**